CRM/Contribute/BAO/Contribution.php

```diff
@@ -3609,7 +3609,9 @@ public static function recordFinancialAccounts(&$params, $financialTrxnValues =
         'fee_amount' => $params['fee_amount'] ?? NULL,
         'net_amount' => CRM_Utils_Array::value('net_amount', $params, $totalAmount),
         'currency' => $params['contribution']->currency,
-        'trxn_id' => $params['contribution']->trxn_id,
+        'trxn_id' => $params['trxn_id'],
```

Comment:
`$params['contribution']->trxn_id` and `$params['trxn_id']` are the same at this point so it makes no difference. However, it seems more *correct* to take the `trxn_id` from the `$params` array as we are recording a payment/financialtrxn record at this point and not a contribution. This should make it slightly more straightforward as we further refactor this code.